src/FormattingSettingsService.ts

```diff
@@ -50,74 +57,115 @@ export class FormattingSettingsService implements IFormattingSettingsService {
         let formattingModel = {
             cards: []
         }
-
-        formattingSettingsModel.cards?.forEach((card: Card) => {
-            if (!card)
-                return;
-
-            const objectName = card.name;
-            const groupUid = card.name + "-group";
-
-            let formattingGroup: visuals.FormattingGroup = {
-                displayName: undefined,
-                slices: [],
-                uid: groupUid
-            }
-
-            let formattingCard: visuals.FormattingCard = card.getFormattingCard(objectName, formattingGroup, this.localizationManager);
-
-            formattingModel.cards.push(formattingCard);
-
-            // In case formatting model adds data points or top categories (Like when you modify specific visual category color).
-            // these categories use same object name and property name from capabilities and the generated uid will be the same for these formatting categories properties
-            // Solution => Save slice names to modify each slice uid to be unique by adding counter value to the new slice uid
-            const sliceNames: { [name: string]: number } = {};
-
-            // Build formatting container slice for each property
-            if (card.container) {
-                const container = card.container;
-                const containerUid = groupUid + "-container";
-                const formattingContainer: visuals.FormattingContainer = {
-                    displayName: (this.localizationManager && container.displayNameKey)
-                        ? this.localizationManager.getDisplayName(container.displayNameKey) : container.displayName,
-                    description: (this.localizationManager && container.descriptionKey)
-                        ? this.localizationManager.getDisplayName(container.descriptionKey) : container.description,
-                    containerItems: [],
-                    uid: containerUid
+        
+        formattingSettingsModel.cards
+            .filter((card: Card) => card.visible ?? true)
+            .forEach((card: Card) => {
+                if (!card)
```

Comment:
I think we can get rid of this check. It's not possible to have the card empty/undefined in forEach loop as it would throw an error on the filter stage trying to get `undefined.visible`
If it's possible to have the card undefined, then we should move the condition to filter callback. For example: `card && card.visible ?? true`

Reply:
It is indeed not possible to have a `undefined` card, so the check is indeed obsolete.